src/Core/Services/Implementations/UserService.cs

```diff
@@ -1087,13 +1087,24 @@ public async Task<bool> TwoFactorProviderIsEnabledAsync(TwoFactorProviderType pr
             return await CanAccessPremium(user);
         }
 
+        //TODO refactor this to use the below method and enum
         public async Task<string> GenerateEnterprisePortalSignInTokenAsync(User user)
         {
             var token = await GenerateUserTokenAsync(user, Options.Tokens.PasswordResetTokenProvider,
                 "EnterprisePortalTokenSignIn");
             return token;
         }
 
+
+        public async Task<string> GenerateSignInTokenAsync(User user, TokenPurpose purpose)
+        {
+            var token = await GenerateUserTokenAsync(
+                user, 
+                Options.Tokens.PasswordResetTokenProvider,
+                purpose.ToString());
+            return token;
+        }
```

Comment:
I would get rid of the enum and use a string like the others. You could create a static class of const strings if you want to re-use them. Some purposes need more data than a static string. See `SendTwoFactorEmailAsync` for example.

Reply:
Any preference on where that static class would live? I see a Constants.cs file, though it is mostly empty. Of course there is also StaticStore, but I'm not sure if this would belong there or not.

Reply:
Sure, Constants works

Reply:
Done